src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp

```diff
@@ -489,9 +535,32 @@ bool TypeLookupManager::create_endpoints()
     else
     {
         EPROSIMA_LOG_ERROR(TYPELOOKUP_SERVICE, "Typelookup request writer creation failed.");
-        delete builtin_request_writer_history_;
-        builtin_request_writer_history_ = nullptr;
-        return false;
+        ret = false;
+    }
+
+    // Built-in request reader
+    if (nullptr == request_listener_)
+    {
+        request_listener_ = new TypeLookupRequestListener(this);
+    }
```

Comment:
This can be removed because it is not going to be nullptr. In master, there are a couple of properties (use_client and use_server) that can be set together or separately. With that configuration it makes sense to check, but with current implementation is obsolete.